packages/table/src/table.tsx

```diff
@@ -681,28 +681,36 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
             selectionModes,
         } = nextProps;
 
-        const newChildArray = React.Children.toArray(children) as Array<React.ReactElement<IColumnProps>>;
+        const hasNewChildren = this.props.children !== nextProps.children;
```

Comment:
if i'm not mistaken, this will be true _every time_ as `children` is a new object reference every time (never `===`)?
perhaps add a `console.log(hasNewChildren)` locally to debug if it's ever `false`?

Reply:
children may be memoized

Reply:
Yes exactly - it needs to be explicitly done by the developer. Memoizing props is a standard part of getting better perf in react so seems reasonable?